src/main/kotlin/org/jetbrains/bio/experiments/fit/Span2PeakCallingExperiment.kt

```diff
@@ -0,0 +1,169 @@
+import org.jetbrains.bio.genome.GenomeQuery
+import org.jetbrains.bio.query.CachingQuery
+import org.jetbrains.bio.query.ReadsQuery
+import org.jetbrains.bio.query.reduceIds
+import org.jetbrains.bio.query.stemGz
+import org.jetbrains.bio.statistics.ClassificationModel
+import org.jetbrains.bio.statistics.Fitter
+import org.jetbrains.bio.statistics.hypothesis.NullHypothesis
+import org.jetbrains.bio.statistics.mixture.PoissonRegressionMixture
+import org.jetbrains.bio.statistics.state.ZLH
+import org.jetbrains.bio.util.div
+import org.jetbrains.bio.viktor.asF64Array
+import java.nio.file.Path
+
+class Span2PeakCallingExperiment<Model : ClassificationModel, State : Any>(
+        genomeQuery: GenomeQuery,
+        paths: SpanDataPaths,
+        mapabilityPath: Path?,
+        fragment: Fragment,
+        binSize: Int,
+        modelFitter: Fitter<Model>,
+        modelClass: Class<Model>,
+        states: Array<State>,
+        nullHypothesis: NullHypothesis<State>,
+        unique: Boolean = true,
+        fixedModelPath: Path? = null
+) : SpanModelFitExperiment<Model, State>(
+    genomeQuery to createDataQuery(genomeQuery, paths, mapabilityPath, fragment, binSize, unique),
+    listOf(paths), mapabilityPath,
+    listOfNotNull("y", paths.control?.let { "input" }),
+    fragment, binSize,
+    modelFitter, modelClass,
+    states, nullHypothesis,
+    unique,
+    fixedModelPath
+) {
+
+    override val modelPath: Path get() = fixedModelPath ?: experimentPath / "$id.span2"
+    override val id = reduceIds(
+        listOfNotNull(paths.treatment, paths.control, mapabilityPath).map { it.stemGz } +
+                listOfNotNull(fragment.nullableInt, binSize).map { it.toString() }
+    )
+
+    companion object {
+
+        fun binnedCoverageAsInt(chr: Chromosome, coverage: Coverage, binSize: Int): IntArray {
+            val len = (chr.length - 1) / binSize + 1
+            val res = IntArray(len)
+            for (i in 0 until len - 1) {
+                res[i] = coverage.getBothStrandsCoverage(
+                    ChromosomeRange(i * binSize, (i + 1) * binSize, chr)
+                )
+            }
+            res[len - 1] = coverage.getBothStrandsCoverage(
+                ChromosomeRange((len - 1) * binSize, chr.length, chr)
+            )
+            return res
+        }
+
+        fun binnedCoverageAsDouble(chr: Chromosome, coverage: Coverage, binSize: Int): DoubleArray {
+            val len = (chr.length - 1) / binSize + 1
+            val res = DoubleArray(len)
+            for (i in 0 until len - 1) {
+                res[i] = coverage.getBothStrandsCoverage(
+                    ChromosomeRange(i * binSize, (i + 1) * binSize, chr)
+                ).toDouble()
+            }
+            res[len - 1] = coverage.getBothStrandsCoverage(
+                ChromosomeRange((len - 1) * binSize, chr.length, chr)
+            ).toDouble()
+            return res
+        }
+
+        fun meanGC(chr: Chromosome, binSize: Int): DoubleArray {
```

Comment:
We already have a class `CpGContent` dedicated to CpG content manipulations, please consider moving this method there and adding tests.

Reply:
Moved the method to `CpGContent`.

Reply:
Added a test.